homeassistant/components/todoist/calendar.py

```diff
@@ -188,6 +196,27 @@ def handle_new_task(call):
         if _due:
             item.update(due=_due)
 
+        _reminder_due: dict = {}
+        if REMINDER_DATE_STRING in call.data:
+            _reminder_due["string"] = call.data[REMINDER_DATE_STRING]
+
+        if REMINDER_DATE_LANG in call.data:
+            _reminder_due["lang"] = call.data[REMINDER_DATE_LANG]
+
+        if REMINDER_DATE in call.data:
+            due_date = dt.parse_datetime(call.data[REMINDER_DATE])
+            if due_date is None:
+                due = dt.parse_date(call.data[REMINDER_DATE])
+                due_date = datetime(due.year, due.month, due.day)
+            # Format it in the manner Todoist expects
+            due_date = dt.as_utc(due_date)
+            date_format = "%Y-%m-%dT%H:%M"
```

Comment:
I know you copied this from above, but as far as I can tell the api rejects this date:

```python
{'error': 'Date is invalid', 'error_code': 480, 'error_extra': {}, 'error_tag': 'INVALID_DATE', 'http_code': 400})
```

If we are going to have time we need to add the seconds both here and above on line 192.
```suggestion
            date_format = "%Y-%m-%dT%H:%M:%S"
```

Reply:
Thanks. I must admit I did not test the date part, just the string part.

One question, I've noticed that parsing is done in UTC. That means that user must provide the time in UTC timezone. Is this desired? On the other hand it might be a bit confusing if user wants to specify time themselves, but on the other hand it might be easier for automations to specify time if we limit ourselves to UTC.

what should we do here?

Reply:
The todoist api expects the date/time to be in UTC, but does allow an additional `timezone` parameter: https://developer.todoist.com/sync/v8/?python#due-dates-with-time-and-fixed-timezone

If you are interested, you are welcome to add timezone support, but may be out of scope for this current pull request.  For the time being, I think it's fine to follow the same pattern that's already in use for due dates.

Reply:
Thanks for comment. This should be fixed now and documentation updated.

Reply:
Thanks for also updating the docs!